backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/InClusterUpgradeValidator.java

```diff
@@ -70,7 +70,7 @@ public ValidationResult isVmReadyForUpgrade(final VM vm) {
         if (vm.getStatus().isSuspended()) {
             errors.addAll(toVmEngineMessage(vm, EngineMessage.CLUSTER_UPGRADE_DETAIL_VM_SUSPENDED));
         }
-        if (!StringUtils.isEmpty(VmCpuCountHelper.isAutoPinning(vm) ? vm.getCurrentCpuPinning() : vm.getCpuPinning())) {
+        if (!StringUtils.isEmpty(VmCpuCountHelper.isDynamicCpuPinning(vm) ? vm.getCurrentCpuPinning() : vm.getCpuPinning())) {
```

Comment:
Why is it not possible to upgrade cluster if a running VM has pinned CPUs?

Reply:
that's a good question - it seems this method tries to find whether vms are bounded to hosts because there was some problem when upgrading from 3.6 (see [this](https://github.com/oVirt/ovirt-engine/commit/c00b80cd1253a0b9ced9bbff2c58b852909eea3b))

Reply:
I didn't touch it, I think it's irrelevant for the current PR. I just preserve the current behavior.

Reply:
> I didn't touch it, I think it's irrelevant for the current PR. I just preserve the current behavior.

sure, unrelated to this PR